src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,56 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+// import { useEffect } from 'react';
+// import { Todo } from '../../types/Todo';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  filter: string,
+  setFilter: React.Dispatch<React.SetStateAction<string>>
+  setSorted: React.Dispatch<React.SetStateAction<string>>
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter:React.FC<Props> = ({
+  filter, setFilter, setSorted,
+}) => {
+  const sortedHandler = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setSorted(e.target.value);
+  };
+
+  return (
+    <form className="field has-addons" onSubmit={e => e.preventDefault()}>
+      <p className="control">
+        <span className="select">
+          <select data-cy="statusSelect" onChange={e => sortedHandler(e)}>
+            <option value="all">All</option>
+            <option value="active">Active</option>
+            <option value="completed">Completed</option>
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          value={filter}
+          onChange={e => setFilter(e.target.value)}
```

Comment:
Good practice to make a handler for this function. To move it to a separate function on use only handlerChangeFilterQuery
` onChange={handlerChangeFilterQuery}`